### PR TITLE
Use infra uid to generate bootstrap kubeconfig

### PR DESCRIPTION
**Pull request by zhujian7**
issue reference: https://issues.redhat.com/browse/ACM-10626

### CODE REVIEW

**Comment by elgnay** [comment]
We may also need to extract the cluster name from the existing bootstrap hub kubeconfig here https://github.com/stolostron/managedcluster-import-controller/pull/346/files#diff-576c58639c3f314f0ece65ca6e525856a23b8d7087214a1653ca35071d07be48L42, and compare it with the UUID of the Infrastructure. Then determine whether we need to re-create the bootstrap hub kubeconfig.

**Comment by zhujian7** [comment]
> We may also need to extract the cluster name from the existing bootstrap hub kubeconfig

@elgnay fixed, PTAL

File: pkg/bootstrap/boostrapkubeconfig.go [hunk 1]
```diff
@@ -181,6 +187,26 @@ func GetKubeAPIServerAddress(ctx context.Context, client client.Client,
 	return "", err
 }
 
+func GetKubeconfigClusterName(ctx context.Context, client client.Client) (string, error) {
+	defaultCluster := "default-cluster"
+	if !helpers.DeployOnOCP {
```
Comment (qiujian16) [inline pkg/bootstrap/boostrapkubeconfig.go:192]: We should add a comment here. We will miss the function if hub is non-ocp, but we are trying to support that.

File: pkg/controller/importconfig/cluster_info.go [hunk 2]
```diff
@@ -74,6 +75,16 @@ func getBootstrapKubeConfigDataFromImportSecret(ctx context.Context, clientHolde
 		return nil, nil, nil
 	}
 
+	// check if the current context cluster name of the bootstrap kubeconfig is changed
+	validCtxClusterName, err := validateContextClusterName(ctx, clientHolder.RuntimeClient, ctxClusterName)
```
Comment (qiujian16) [inline pkg/controller/importconfig/cluster_info.go:79]: note: why we validate here? all generation and validation should be in bootstrap package.

**Review by qiujian16 — COMMENTED** [review]
just some comment should be add. And we need consider some refactor.

File: pkg/controller/importconfig/cluster_info.go (hunk 2 above)
Comment (zhujian7) [reply pkg/controller/importconfig/cluster_info.go:79]: this is to fix YangLe's comment https://github.com/stolostron/managedcluster-import-controller/pull/346#issuecomment-2126303203

File: pkg/bootstrap/boostrapkubeconfig.go (hunk 1 above)
Comment (zhujian7) [reply pkg/bootstrap/boostrapkubeconfig.go:198]: PTAL

File: pkg/controller/importconfig/cluster_info.go (hunk 2 above)
Comment (qiujian16) [reply pkg/controller/importconfig/cluster_info.go:79]: yeah I see. I mean the whole validation func should be put in the bootstrap package.

Comment (zhujian7) [reply pkg/controller/importconfig/cluster_info.go:79]: I will submit a follow-up PR to refactor this, and keep the current PR changes as small as possible to facilitate cherrypick to the old releases.

**Pull request merged**